### DISCUSSION TITLE
Adding hyphen suffix to tokenizer affects prefixes, infixes, and following tokens

### DISCUSSION THREAD

**Discussion by caleb-lindgren** [explosion/spaCy#8153] [Help: Coding & Implementations] [answered]
Following the outline [here](https://spacy.io/usage/linguistic-features#native-tokenizer-additions) in the docs, I tried to customize the tokenizer to split a trailing hyphen as a separate token (e.g. from "yes-" split out "yes" and "-"). However, this led to some strange tokenizations in other cases involving hyphens.

## Default behavior
Here is the expected default behavior of the tokenizer.
```python
import spacy
nlp = spacy.load("en_core_web_sm")

[t for t in nlp("Hi- ex")]    # -> [Hi-, ex]
[t for t in nlp("-12")]       # -> [-12]
[t for t in nlp("20-12")]     # -> [20, -, 12]
```
## Behavior after customization
```python
import spacy
nlp = spacy.load("en_core_web_sm")
suffixes = list(nlp.Defaults.suffixes)
suffixes.append(spacy.lang.char_classes.HYPHENS) # Split trailing hyphen as a token
suffix_re = spacy.util.compile_suffix_regex(suffixes)
nlp.tokenizer.suffix_search = suffix_re.search

[t for t in nlp("Hi- ex")]    # -> [Hi, -, ex] like I wanted
[t for t in nlp("-12")]       # -> [-, 1, 2] Why is "-" split as a prefix, and 
                              #              "12" is split into two tokens?
[t for t in nlp("20-12")]     # -> [20, -, 1, 2]  "12" is split again
```
I don't know why adding hyphens as suffixes is affecting how they're treated when they're prefixes. It's also perplexing that the numbers following the hyphen when it's a prefix or infix are being split into separate tokens for each digit.

## Environment
- **spaCy version:** 3.0.6
- **Platform:** Linux-5.11.16-arch1-1-x86_64-with-arch
- **Python version:** 3.7.7
- **Pipelines:** en_core_sci_lg (0.4.0), en_core_web_sm (3.0.0)

**Comment by adrianeboyd** [comment, marked as answer]
Hi, I think the problem is that `HYPHENS` isn't in the right format for appending to `suffixes`. You want to use `LIST_HYPHENS` instead:

```python
suffixes = list(nlp.Defaults.suffixes) + spacy.lang.char_classes.LIST_HYPHENS
```

Output:

```none
[Hi, -, ex]
[-12]
[20, -, 12]
```

**Reply by caleb-lindgren** [reply]
Oh thanks! Yep, that fixes it. Glad it was just an error on my end, rather than a problem with the package. I guess I didn't follow the pattern from the example in the docs as well as I thought I had :)
